packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -258,7 +258,26 @@ module AlgebraicCombination = {
         : Convolution(convOp)
     }
 
+  let chooseConvolutionOrMonteCarlo = (
+    ~strat: DistributionTypes.asAlgebraicCombinationStrategy,
+    op: Operation.algebraicOperation,
+    t2: t,
+    t1: t,
+  ): result<calculationMethod, error> => {
+    switch strat {
+    | AsDefault => Ok(chooseConvolutionOrMonteCarloDefault(op, t2, t1))
+    | AsConvolution =>
+      switch op {
+      | #Divide | #Power | #Logarithm => Error(RequestedStrategyInvalidError)
+      | (#Add | #Subtract | #Multiply) as convOp => Ok(Convolution(convOp))
+      }
+    | AsMontecarlo => Ok(MonteCarlo)
+    | AsSymbolic => Error(RequestedStrategyInvalidError)
```

Comment:
it seems as if your code, when asking for convolution, will first check if a symbolic solution is available and do that if it is. Might be worth changing the calculationMethod type to include symbolic and move the analytical choice code into this method?

Reply:
+1, I think this logic is in the wrong place.